js/components/main.js

```diff
@@ -482,37 +483,46 @@ class Main extends ImmutableComponent {
     currentWindow.on('maximize', function () {
       windowActions.setMaximizeState(true)
     })
-
     currentWindow.on('unmaximize', function () {
       windowActions.setMaximizeState(false)
     })
-
+    // NOTE: setTimeout is used to avoid spamming events
+    let resizeTimeout = null
     currentWindow.on('resize', function (event) {
-      windowActions.saveSize(event.sender.getSize())
+      if (resizeTimeout) {
+        clearTimeout(resizeTimeout)
+      }
+      resizeTimeout = setTimeout(function () {
+        const size = event.sender.getSize()
+        // NOTE: the default window size is whatever the last window resize was
+        appActions.defaultWindowParamsChanged(size)
+        windowActions.saveSize(size)
```

Comment:
I think with the updated design that we'll move to (where app state has control over window state), it would be a no brainer as one action. 

With out system now, does the WindowStore also receive AppStore events? (if so, maybe I can add `id` in there and then do a lookup in the handler from WindowStore)